### PR TITLE
docs: add pointer to caddy config for docker compose http/https

### PR DESCRIPTION
**Pull request by ggilmore**
Reformats our NGINX docs to Nginx / Caddy agnostic and adds a pointer to the docker-compose specific instructions

### CODE REVIEW

**Comment by ggilmore** [comment]
@slimsag @dadlerj Please re-review - updated

**Comment by ggilmore** [comment]
Merging, please review afterwards

File: doc/admin/nginx.md [hunk 1]
```diff
@@ -1,138 +1,7 @@
-# Sourcegraph NGINX HTTP and HTTPS/SSL configuration
+---
+ignoreDisconnectedPageCheck: true
+---
 
-In Sourcegraph, [NGINX](https://www.nginx.com/resources/glossary/nginx/) acts as a [reverse proxy](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/) for the Sourcegraph front-end server, meaning NGINX proxies external HTTP (and [HTTPS](#nginx-ssl-https-configuration)) requests to the Sourcegraph front-end.
+# Sourcegraph Nginx sHTTP and HTTPS/SSL configuration
```
Comment (dadlerj) [inline doc/admin/nginx.md]: Did a lowercase "s" sneak into this?

**Comment by dadlerj** [comment]
Awesome, thanks!!

File: doc/admin/http_https_configuration.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+Sourcegraph's single Docker image and Kubernetes deployments use [NGINX](https://www.nginx.com/resources/glossary/nginx/) as a [reverse proxy](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/) for the Sourcegraph front-end server, meaning NGINX proxies external HTTP (and [HTTPS](#nginx-ssl-https-configuration)) requests to the Sourcegraph front-end.
+
+![NGINX and Sourcegraph architecture](img/sourcegraph-nginx.svg)
+
+**Note**: Non-sighted users can view a [text-representation of this diagram](sourcegraph-nginx-mermaid.md).
+
+### Sourcegraph single instance (Docker)
+
+<!-- TODO(ryan): Change heading to ## The default `nginx.conf` file and how to extend/override default configuration and add section
```
Comment (dadlerj) [inline doc/admin/http_https_configuration.md]: This can be deleted!

File: doc/admin/http_https_configuration.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
+# Sourcegraph HTTP and HTTPS/SSL configuration
```
Comment (dadlerj) [inline doc/admin/http_https_configuration.md]: Given how long this file is, adding a short agenda at the top with three bullets ("Docker image/Cluster-NGINX", "Docker Compose", and "Other clusters") would be helpful here. E.g., see https://docs.sourcegraph.com/admin/auth for an example of what I mean.

**Pull request merged**